include/kernel.h

```diff
@@ -3714,6 +3714,32 @@ __syscall int k_mutex_unlock(struct k_mutex *mutex);
  * @}
  */
 
+
+
+struct k_condvar {
+	_wait_q_t wait_q;
+};
+
+#define Z_CONDVAR_INITIALIZER(obj) \
+	{ \
+	.wait_q = Z_WAIT_Q_INIT(&obj.wait_q), \
+	}
+
+int k_condvar_init(struct k_condvar *condvar);
+
+int k_condvar_signal(struct k_condvar *condvar);
+
+int k_condvar_broadcast(struct k_condvar *condvar);
+
+int k_condvar_wait(struct k_condvar *condvar,
+		   struct k_mutex *mutex,
+		   k_timeout_t timeout);
+
+
+#define K_CONDVAR_DEFINE(name) \
+	Z_STRUCT_SECTION_ITERABLE(k_condvar, name) = \
```

Comment:
don't see any need for iteration at the moment, will this be added later?

Reply:
for object tracing eventually, but probably will drop it.